### PR TITLE
feat: generate reference docs

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Addresses #206.  

[TypeDoc](https://github.com/TypeStrong/typedoc) has a ton of un-merged PRs, so I had to go create my own fork, fix it, and publish to npm.  That's why it's `@justinbeckwith/TypeDoc`.

I have it set up so you can run `npm run generate-docs`, and `npm run publish-docs` to push it to GitHub pages.  

You can see the generated docs here:
http://google.github.io/google-auth-library-nodejs/

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
cc @broady

**Comment by JustinBeckwith** [comment]
@google/google-node-team can someone take a look?

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "lodash.isstring": "^4.0.1"
   },
   "devDependencies": {
+    "@justinbeckwith/typedoc": "^0.10.1",
```
Comment (jinwoo) [inline package.json]: You published typedoc under your name? How is it different from the `typedoc` package?

**Comment by jinwoo** [comment]
And what's up with the cla error? The commits are only from you, right?

File: package.json (hunk 1 above)
Comment (JustinBeckwith) [reply package.json]: Yup, covered this above. TypeDoc has a ton of un-merged PRs, and it doesn't work today with TypeScript 2.6.  I had to go create my own fork, update a bunch of dependencies, add TypeScript 2.6 support, and publish to npm. That's why it's @justinbeckwith/TypeDoc :)  When they get their issues sorted out, we should try to move back.

**Comment by JustinBeckwith** [comment]
On CLA bot - I think it's been a little janky lately :) Only commit here is by me.

File: package.json (hunk 1 above)
Comment (jinwoo) [reply package.json]: Got it. BTW https://www.npmjs.com/package/@justinbeckwith/typedoc points to https://github.com/TypeStrong/TypeDoc as its repo, not yours. Is it intentional?

**Comment by JustinBeckwith** [comment]
Nope, oversight on my part.  Was trying to make only the needed changes :) . I will go back and update that though.

**Pull request merged**